gateway-service/src/main/java/com/ca/mfaas/gateway/routing/MfaasRouteLocator.java

```diff
@@ -86,7 +87,20 @@
                     log.error("Cannot find any instances of service: " + serviceId);
                     return null;
                 }
-                List<String> keys = createRouteKeys(serviceInstances, routedServices, serviceId);
+
+                //serviceId is converted to lowercase by Discovery Service. Recover serviceId which contains uppercase characters
```

Comment:
Could you, please, move this block to separate method? Like this:
```java
    private String getOriginalServiceId(String serviceId, List<ServiceInstance> serviceInstances) {
        //serviceId is converted to lowercase by Discovery Service. Recover serviceId which contains uppercase characters
        String originalServiceId = serviceId;
        if (serviceInstances.size() > 0) {
            ServiceInstance instance = serviceInstances.get(0);
            if (instance instanceof EurekaDiscoveryClient.EurekaServiceInstance) {
                String vipAddr = ((EurekaDiscoveryClient.EurekaServiceInstance) instance).getInstanceInfo().getVIPAddress();
                if (serviceId.equalsIgnoreCase(vipAddr)) {
                    originalServiceId = vipAddr;
                }
            }
        }
        return originalServiceId;
    }
```
And put the calling of it directly in to calling of `createRouteKeys` method, like this: 
```java
List<String> keys = createRouteKeys(serviceInstances, routedServices, getOriginalServiceId(serviceId, serviceInstances));
```

Reply:
Thank you for the comments, I've pushed a commit, and modified the code as your comments, thank you